File: Problems/No.121_Best_Time_to_Buy_and_Sell_Stock/best_time_to_buy_and_sell_stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        // parameters
        int max_profit = 0;
        int price_buy = prices[0];
        
        // start compute
        for (int i=0; i<prices.size(); i++) {
            
            price_buy = min(price_buy, prices[i]);
            max_profit = max(max_profit, prices[i]-price_buy);
            
        }
        
        return max_profit;
        
    }
};

File: Problems/No.121_Best_Time_to_Buy_and_Sell_Stock/best_time_to_buy_and_sell_stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        # parameters
        max_profit = 0
        buy_price = prices[0]
        
        # start compute
        for price in prices:
            buy_price = min(buy_price, price)
            max_profit = max(max_profit, price-buy_price)
        
        return max_profit
